plugins/account_history_plugin/account_history_plugin.cpp

```diff
@@ -283,45 +283,59 @@ vector<account_name> account_history_plugin_impl::get_controlled_accounts(const
    return vector<account_name>(accounts.begin(), accounts.end());
 }
 
-void account_history_plugin_impl::applied_block(const signed_block& block)
+static vector<account_name> generated_affected_accounts(const chain::transaction_trace& trx_trace) {
+   vector<account_name> result;
+   for (const auto& at: trx_trace.action_traces) {
+      for (const auto& auth: at.act.authorization) {
+         result.emplace_back(auth.actor);
+      }
+
+      result.emplace_back(at.receiver);
+   }
+
+   fc::deduplicate(result);
+   return result;
+}
+
+void account_history_plugin_impl::applied_block(const chain::block_trace& trace)
 {
+   const auto& block = trace.block;
    const auto block_id = block.id();
    auto& db = chain_plug->chain().get_mutable_database();
    const bool check_relevance = filter_on.size();
-   for (const signed_transaction& trx : block.input_transactions)
+   auto process_one = [&](const chain::transaction_trace& trx_trace )
    {
-      if (check_relevance && !is_scope_relevant(trx.read_scope) && !is_scope_relevant(trx.write_scope))
-         continue;
+      auto affected_accounts = generated_affected_accounts(trx_trace);
+      if (check_relevance && !is_scope_relevant(affected_accounts))
+         return;
 
-      auto trx_obj_ptr = db.find<transaction_history_object, by_trx_id>(trx.id());
+      auto trx_obj_ptr = db.find<transaction_history_object, by_trx_id>(trx_trace.id);
       if (trx_obj_ptr != nullptr)
-         continue; // on restart may already have block
+         return; // on restart may already have block
 
-      db.create<transaction_history_object>([&block_id,&trx](transaction_history_object& transaction_history) {
+      db.create<transaction_history_object>([&block_id,&trx_trace](transaction_history_object& transaction_history) {
          transaction_history.block_id = block_id;
-         transaction_history.transaction_id = trx.id();
+         transaction_history.transaction_id = trx_trace.id;
+         transaction_history.transaction_status = trx_trace.status;
```

Comment:
Do we need to add functionality around transaction_status now, or is it something that is already written up to be added?

Reply:
I tried to make the minimal changes necessary, we may want some indexing around transaction status as part of a future issue.